app/src/main/java/org/oppia/app/profile/AdminAuthActivityPresenter.kt

```diff
@@ -53,17 +57,43 @@ class AdminAuthActivityPresenter @Inject constructor(
         return@setOnClickListener
       }
       if (inputPin == adminPin) {
-        activity.startActivity(
-          AddProfileActivity.createAddProfileActivityIntent(
-            context, activity.intent.getIntExtra(KEY_ADMIN_AUTH_COLOR_RGB, -10710042)
-          )
-        )
+        when (activity.intent.getIntExtra(KEY_ADMIN_AUTH_ENUM, 0)) {
+          AdminAuthEnum.PROFILE_ADMIN_CONTROLS.value -> {
+            activity.startActivity(
+              AdministratorControlsActivity.createAdministratorControlsActivityIntent(
+                context, activity.intent.getIntExtra(KEY_ADMIN_AUTH_PROFILE_ID, -1)
+              )
+            )
+          }
+          AdminAuthEnum.PROFILE_ADD_PROFILE.value -> {
+            activity.startActivity(
+              AddProfileActivity.createAddProfileActivityIntent(
+                context, activity.intent.getIntExtra(KEY_ADMIN_AUTH_COLOR_RGB, -10710042)
+              )
+            )
+          }
+        }
       } else {
         authViewModel.errorMessage.set(activity.resources.getString(R.string.admin_auth_incorrect))
       }
     }
   }
 
+  private fun setTitleAndSubTitle(binding: AdminAuthActivityBinding?) {
+    when (activity.intent.getIntExtra(KEY_ADMIN_AUTH_ENUM, 0)) {
+      AdminAuthEnum.PROFILE_ADMIN_CONTROLS.value -> {
+        binding?.adminAuthToolbar?.title = context.resources.getString(R.string.administrator_controls)
+        binding?.headingText?.text = context.resources.getString(R.string.admin_auth_heading)
+        binding?.subText?.text = context.resources.getString(R.string.admin_auth_admin_controls_sub)
+      }
+      AdminAuthEnum.PROFILE_ADD_PROFILE.value -> {
+        binding?.adminAuthToolbar?.title = context.resources.getString(R.string.add_profile_title)
+        binding?.headingText?.text = context.resources.getString(R.string.admin_auth_heading)
```

Comment:
I think this heading should be `Add Profile`. Right?

Reply:
No, as per the mock, "Add Profile" is in Toolbar not in headingText.